dimod/binary/binary_quadratic_model.py

```diff
@@ -1018,11 +1018,20 @@ def energy(self, sample, dtype: Optional[DTypeLike] = None) -> Bias:
         energy, = energies
         return energy
 
-    def flip_variable(self, v: Hashable):
+    def flip_variable(self, v: Variable):
         """Flip the specified variable in a binary quadratic model."""
-        for u in self.adj[v]:
```

Comment:
Good catch.

Reply:
It actually worked before because we were modifying `bqm.spin` which is a "virtual" BQM of the SPIN vartype. E.g.
```python
In [5]: bqm = dimod.BQM('BINARY')

In [6]: bqm.spin.set_quadratic('a', 'b', -1)

In [7]: bqm
Out[7]: BinaryQuadraticModel({'a': 2.0, 'b': 2.0}, {('b', 'a'): -4.0}, -1.0, 'BINARY')
```
but that machinery is relatively expensive/slow which is why I changed it here.